From 0b3497731b911736d436f60e34a099331c5a3829 Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Fri, 2 Nov 2018 17:20:26 +0530
Subject: [PATCH 0130/1851] ASoC: xlnx: Disable clock wizard usage in audio
 driver

commit 278e1e77bfcf170939fc46471db6454eea08830b from
https://github.com/Xilinx/linux-xlnx.git

Current solution which uses clocking wizard, couldn't generate
accurate clock due to inability of clock wizard driver to program
fractional value of clock divider value. This is resulting in audio
loss in a simulataneous capture and playback audio usecase.
This patch disables its usage.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Sandip Kothari <sandipk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_pl_snd_card.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_pl_snd_card.c b/sound/soc/xilinx/xlnx_pl_snd_card.c
index 9024db858f9b..1fd3c9811159 100644
--- a/sound/soc/xilinx/xlnx_pl_snd_card.c
+++ b/sound/soc/xilinx/xlnx_pl_snd_card.c
@@ -148,11 +148,6 @@ static int xlnx_i2s_card_hw_params(struct snd_pcm_substream *substream,
 		}
 	}
 
-	prv->mclk_val = prv->mclk_ratio * sample_rate;
-	ret = clk_set_rate(prv->mclk, prv->mclk_val);
-	if (ret)
-		return ret;
-
 	clk_div = DIV_ROUND_UP(prv->mclk_ratio, 2 * ch * data_width);
 	ret = snd_soc_dai_set_clkdiv(cpu_dai, 0, clk_div);
 
@@ -213,7 +208,6 @@ static struct snd_soc_dai_link xlnx_snd_dai[][XLNX_MAX_PATHS] = {
 		{
 			.name = "xilinx-hdmi-playback",
 			SND_SOC_DAILINK_REG(xlnx_hdmi_tx),
-			.ops = &xlnx_hdmi_card_ops,
 		},
 		{
 			.name = "xilinx-hdmi-capture",
@@ -224,7 +218,6 @@ static struct snd_soc_dai_link xlnx_snd_dai[][XLNX_MAX_PATHS] = {
 		{
 			.name = "xlnx-sdi-playback",
 			SND_SOC_DAILINK_REG(xlnx_sdi_tx),
-			.ops = &xlnx_sdi_card_ops,
 		},
 		{
 			.name = "xlnx-sdi-capture",
-- 
2.31.1

